### PR TITLE
Feat email extra params

### PR DESCRIPTION
**Pull request by fanatic75**
## What does this PR do?

PR for extra params in email message

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/utopia-php/messaging/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/Utopia/Messaging/Adapter/Email/Mailgun.php [hunk 1]
```diff
@@ -45,21 +45,32 @@ protected function process(EmailMessage $message): string
...
+        $body = [
+            'to' => \implode(',', $message->getTo()),
+            'from' => "{$message->getFrom()}<{$message->getSenderEmailAddress()}>",
+            'subject' => $message->getSubject(),
+            'text' => $message->isHtml() ? null : $message->getContent(),
+            'html' => $message->isHtml() ? $message->getContent() : null,
+        ];
+
+        if (! \is_null($message->getCcName()) && ! \is_null($message->getCcEmail())) {
+            $body['cc'] = "{$message->getCcName()}<{$message->getCcEmail()}>";
```
[the remark this answers is not in the document]
Comment (fanatic75) [reply src/Utopia/Messaging/Adapter/Email/Mailgun.php]: made it to array

[the remark this answers is not in the document]
Comment (fanatic75) [reply src/Utopia/Messaging/Adapter/Email/Mailgun.php]: yes, it shows the name of the cc with the  email address

File: src/Utopia/Messaging/Messages/Email.php [hunk 2]
```diff
@@ -10,18 +10,33 @@ class Email implements Message
...
      * @param  array<string, mixed>|null  $attachments The attachments of the email.
      * @param  bool  $html Whether the message is HTML or not.
      */
     public function __construct(
         private array $to,
         private string $subject,
         private string $content,
-        private ?string $from = null,
+        private string $from,
+        private string $senderEmailAddress,
```
[the remark this answers is not in the document]
Comment (fanatic75) [reply src/Utopia/Messaging/Messages/Email.php]: done this

File: src/Utopia/Messaging/Messages/Email.php [hunk 3]
```diff
@@ -42,11 +57,41 @@ public function getContent(): string
...
+    {
+        return $this->senderEmailAddress;
+    }
+
+    public function getReplyTo(): string
+    {
+        return $this->replyTo;
+    }
+
+    public function getCcName(): ?string
```
[the remark this answers is not in the document]
Comment (fanatic75) [reply src/Utopia/Messaging/Messages/Email.php]: updated to getCC and getBCC

**Review by abnegate — APPROVED** [review]

**Pull request merged**